### PR TITLE
Applied Template

### PR DESCRIPTION
**Pull request by nabobalis**
Latest version now with a run of the linters I like on top. 

No more travis but means circle and azure need setting up.

### CODE REVIEW

File: ez_setup.py [hunk 1]
```diff
@@ -54,7 +54,9 @@ def _python_cmd(*args):
```
Comment (Cadair) [inline ez_setup.py]: this can go, why is this in the template :nauseated_face:

Comment (nabobalis) [reply ez_setup.py]: It wasn't deleted?

File: irispy/extern/__init__.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
```
Comment (Cadair) [inline irispy/extern/__init__.py]: Do we need to be writing an extern if we don't have anything to put it to it? is it a cookie option?

Comment (nabobalis) [reply irispy/extern/__init__.py]: There is a c extern option but not a normal extern one.

**Comment by nabobalis** [comment]
It seems that not all the changes I was able to copy over.

**Comment by nabobalis** [comment]
TODO: SQUASH THE COMMITS, FIX CIRCLECI, Probably some more

**Comment by nabobalis** [comment]
I think this is now good to go at least for a merge and fix issues after point of view.

**Pull request closed without merging**